src/Mvc/Mvc.Core/src/ServiceFilterAttribute.cs

```diff
@@ -48,7 +55,16 @@ public IFilterMetadata CreateInstance(IServiceProvider serviceProvider)
     {
         ArgumentNullException.ThrowIfNull(serviceProvider);
 
-        var filter = (IFilterMetadata)serviceProvider.GetRequiredService(ServiceType);
+        IFilterMetadata filter;
+        if (string.IsNullOrWhiteSpace(Key as string))
```

Comment:
The Key may not be a string, it can be any object! Please copy the code the right way!